### ISSUE TITLE
[NPM] Packages without defined repository

### ISSUE THREAD

**Issue by Lykathia** [webjars/webjars#1038]
Not sure how to handle these ones, as ... well, it gets hard to find their upstream to fix :(

These are all of the form `JsResultException(errors:List((/repository/url,List(ValidationError(error.path.missing,WrappedArray()),`.

```
indexof: 0.0.1
has-binary: 0.1.6
parseuri: 0.0.2
object-component: 0.0.3
has-binary-data: 0.1.3
```

**Comment by jamesward** [comment]
Wow.  Not quite sure what to do about this.

Any ideas?

**Comment by Lykathia** [comment]
These repositories are all dependencies of socket.io. Many of them are part of https://github.com/automattic

In this instance, could potentially fix upstream. But where NPM doesn't seem to enforce the need for a repository to be listed, is there any way we could use the npm homepage as a fallback homepage if the repo is missing?

**Comment by jamesward** [comment]
It looks like at a minimum we need a VCS URL:
https://bintray.com/docs/api/#_create_package

Lets first try to get this fixed upstream.  If that fails we can figure out a plan b.

**Comment by Lykathia** [comment]
I think I've passed my breaking point. Too much of a quagmire trying to get package dependencies all figured out. NPM just seems too ... loose ... in what they allow/support.

If anyone stumbles across these particulars in the future:

[indexof](https://github.com/component/indexof) Fixed in 0.0.3, but not published.
[has-binary](https://github.com/Automattic/has-binary)
[parseuri](https://github.com/get/parseuri) Fixed in 0.0.3+
[object-component](https://github.com/component/object) Fixed in repo, not published. Also missing license.
[has-binary-data](https://github.com/kevin-roark/has-binarydata.js)

**Comment by jamesward** [comment]
Thanks @Lykathia for wading through that quagmire!  Not fun.  But the JS ecosystem is better for your effort.  :)

I'm going to close this.  If there is anything that we can do to address these issues, please open another issue.
